src/g_editor.c

```diff
@@ -3718,6 +3719,37 @@ static void canvas_copy(t_canvas *x)
     }
 }
 
+char *escape_clipboard_special_chars(const char *str) {
+    int i, j = 0, len = strlen(str);
+    char *result = (char *) malloc(2 * len + 1);
+    if(!result) return NULL;
+    
+    for (i = 0; i < len; i++) {
+        if (str[i] == '\"' || str[i] == '[' || str[i] == ']') {
+            result[j++] = '\\';
+        }
+        result[j++] = str[i];
+    }
+    result[j] = '\0';
+    return result;
+}
+
+static void canvas_copy_to_clipboard_as_text(t_canvas *x)
+{
+    t_binbuf *bb = canvas_docopy(x);
+    if (!bb)
+        return;
+    char *buf;
+    int ac;
+    binbuf_gettext(bb, &buf, &ac);
+    char *escaped_buf = escape_clipboard_special_chars(buf);
+    sys_vgui("clipboard clear\n");
+    sys_vgui("clipboard append \"%s\"\n", escaped_buf);
```

Comment:
urgh no.
please use `pdgui_vmess()` instead.

Reply:
Thanks. Still need to check the other way around, if I can make it work using existing methods when pdsending from tcl